### ISSUE TITLE
Urgent: Any plans to update activedirectory2 to support LDAPjsv3

### ISSUE THREAD

**Issue by tfrancois** [jsumners/node-activedirectory#99]
I currently have multiple projects that leverage this library which lists LDAPjs as a dependency.  With the release of LDAPjsv3 I'm worried that I will not be able to update my project to use v3 of LDAPjs which I reference separately in my project for custom functionality that activeDirectory2 doesn't provide to my exact specialized requirements.  My projects are lean and I flatten dependencies using flatten-packages (for example) and therefore may create some conflict here if ad2 doesn't support ldapjsv3 out of the box.

Any help here @jsumners?  I depend quite heavily on your excellent libraries and want to make sure I can continue to use them as I have in the past.  Please advise - and thank you.

**Comment by jsumners** [comment]
No, I do not have any plans or motivation to update this library.

It will be very difficult to do so, because the test suite for `activedirectory2` monkey patches the `ldapjs` server code quite heavily -- https://github.com/jsumners/node-activedirectory/tree/8ff17bdf366a2d6926879ba06fbe84ba0171c01f/test/mockServer.

Any attempt to update this library should start with a complete revamp of the test suite to use some sort of Docker-ized ActiveDirectory (or ActiveDirectory-like) instance. In invested about 45 minutes researching such a container this morning and reached an impasse described in https://github.com/ldapjs/node-ldapjs/issues/828#issuecomment-1443699228.

Without any financial motivation, I consider this project at the end of its life.

**Comment by tfrancois** [comment]
Hi @jsumners.  Thank you for the response.  Sir if I'm to understand the issue with moving forward, you need the ability to be able to rest against an AD or dockerized instance to even consider moving forward, am I right?

If so, is the following link I found something worth looking into?  If it is, I would be willing to explore implementing one as the document describes in order to facilitate this testing:
https://schneide.blog/2022/06/27/running-a-containerized-activedirectory-for-developers/

If that would not suffice, I have a completely non-production, development only instance of AD that is hosted on AWS that I would gladly make available to you for testing.

And lastly, if the last hurdle is the need to entice you with a financial consideration to move forward, how much would that take?  Please feel free to message me privately (if you'd rather not make it public) if this is something you'd be open to doing and how much would something like that cost?

This library is the gold-standard for development against MS AD in my opinion and still widely used by developers based on the weekly download numbers.  To not be able to take advantage of the changes you've made in LDAPv3 would be a travesty.  No pressure, lol.  Please know the time and effort you took to develop AD2 and its unique way to handle very large directories so easily is priceless - and at some point I know will be something that will see an EOL, but maybe before that time comes we could extend its life dramatically by at least allowing it to support LDAPv3 and all the improvements made there.  I'm willing to help in whatever way is necessary to make that happen.  Hope you're willing to as well.  Thanks again!

**Comment by jsumners** [comment]
We can't rely on external resources being available for the lifetime of the project. So, thank you for offering such resources, but we can't use them. 

That being said, some preliminary research indicates that a samba container is viable. Please email me via `oss.consulting@zifbang.com` and we can discuss a sponsorship commission for this work.

**Comment by tfrancois** [comment]
Awesome!  Just sent you an email following up for more details.  Thanks again.

**Comment by jsumners** [comment]
> Awesome!  Just sent you an email following up for more details.  Thanks again.

I replied the following day. Please check your spam to make sure it didn't get delivered there.
